src/i18_bluesky/plans/align_lookup_2.py

```diff
@@ -0,0 +1,127 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+import bluesky.plan_stubs as bps
+import numpy as np
+import sympy as sp
+from bluesky.devices.monochromator import DCM
+from dodal.common import MsgGenerator, inject
+
+monochromator = inject("monochromator")
+undulator = inject("undulator")
+pinhole = inject("pinhole")
+KBMirror = inject("kb_mirror")
+
+
+def calculate_derivative_maxima(data: np.ndarray) -> List[float]:
+    x = sp.Symbol("x")
+    y = sp.interpolating_spline(3, sp.lambdify(x, data))
+    derivative = y.diff(x)
+    critical_points = sp.solve(derivative, x)
+    return [float(cp) for cp in critical_points if cp.is_real]
+
```

Comment:
x = motor position
y = transmission strength

presumably currently y = data, but we need to be passing also the values of x, so data could be array[tuple(float, float)] or otherwise passing in x?